src/parser.rs

```diff
@@ -3,38 +3,80 @@ use crate::filesystem;
+    let parts = shellwords::split(text)
+        .map_err(|_| anyhow!("Given options are missing a closing quote"))?;
+
+    parts
+        .into_iter()
+        .filter(|part| {
+            // We'll take parts in pairs of 2: (argument, value). Flags don't have a value tho, so we filter and handle them beforehand.
+            match part.as_str() {
+                "--multi" => {
+                    multi = true;
+                    false
+                }
+                "--prevent-extra" => {
+                    prevent_extra = true;
+                    false
+                }
+                _ => true,
             }
-            "--column" => opts.column = Some(parts.next().unwrap().parse::<u8>().unwrap()),
-            "--delimiter" => opts.delimiter = Some(parts.next().unwrap().to_string()),
-            "--query" => opts.query = Some(parts.next().unwrap().to_string()),
-            "--filter" => opts.filter = Some(parts.next().unwrap().to_string()),
-            "--preview" => opts.preview = Some(parts.next().unwrap().to_string()),
-            "--preview-window" => opts.preview_window = Some(parts.next().unwrap().to_string()),
-            "--header" => opts.header = Some(parts.next().unwrap().to_string()),
-            "--overrides" => opts.overrides = Some(parts.next().unwrap().to_string()),
-            _ => (),
-        }
-    }
+        })
+        .collect::<Vec<_>>()
+        .chunks(2)
+        .map(|flag_and_value| {
+            if let [flag, value] = flag_and_value {
+                match flag.as_str() {
+                    "--headers" | "--header-lines" => {
+                        opts.header_lines = value
+                            .parse::<u8>()
+                            .context("Value for `--headers` is invalid u8")?
+                    }
+                    "--column" => {
+                        opts.column = Some(
+                            value
+                                .parse::<u8>()
+                                .context("Value for `--column` is invalid u8")?,
+                        )
+                    }
+                    "--delimiter" => opts.delimiter = Some(value.to_string()),
+                    "--query" => opts.query = Some(value.to_string()),
+                    "--filter" => opts.filter = Some(value.to_string()),
+                    "--preview" => opts.preview = Some(value.to_string()),
+                    "--preview-window" => opts.preview_window = Some(value.to_string()),
+                    "--header" => opts.header = Some(value.to_string()),
+                    "--overrides" => opts.overrides = Some(value.to_string()),
+                    _ => (),
+                }
+                Ok(())
+            } else if let [flag] = flag_and_value {
+                Err(anyhow!("No value provided for the flag `{}`", flag))
+            } else {
+                unreachable!() // Chunking by 2 allows only for tuples of 1 or 2 items...
```

Comment:
Why not use an else in the line 72 instead?

Reply:
The `chunks(2)` call on :46 returns slices (`&[..]`) of the collection. The problem is that, despite telling `chunks` that the slices should be `2` long, it doesn't know this at compile time. So Rust wants us to handle every possible case of slice length exhaustively.

Concrete this means that Rust wants the following let binding to be exhaustive.
```rust
} else {
    let [flag] = flag_and_value;
    Err(anyhow!("No value provided for the flag `{}`", flag))
}
```
![image](https://user-images.githubusercontent.com/25766329/77332180-01848880-6d22-11ea-9fc0-48b083890a88.png)

You could do the following but obviously it wouldn't be as helpful an error:
```rust
} else {
    Err(anyhow!("No value provided for a flag"))
}
```